src/types.ts

```diff
@@ -3,7 +3,7 @@ export type Severity = 'info' | 'low' | 'moderate' | 'high' | 'critical';
 export type SeverityCounts = Record<Severity, number>;
 export type SeverityCountsWithTotal = Record<Severity | 'total', number>;
 
-export type Advisories = Record<string, Advisory>;
+export type Advisories = Record<string, Npm6Advisory>;
```

Comment:
🐘 
this is probably my lack of Js knowledge, what is going on with the code using `< >`? is it some kind of function call?

Reply:
this repo is written in typescript - javascripts strongly typed cousin

Reply:
It's a [generic](https://www.typescriptlang.org/docs/handbook/generics.html) - this is actually a TypeScript thing, which is a superset of Javascript.

Type code like this only "exists" at compile type - the runtime code is regular JS (which this is not, so its removed by the compiler).

Specifically, generics are like parameters for types - they let you combine types together without having to know all the details of what a user is passing in.

A great classic example is arrays, which have type `Array<T>` - You don't actually care what type `T` is, your array methods (`push`, `shift`, `slice`, `pop`, `size`, etc) will work regardless of what the array actually holds.

So the "type" for an Array class looks something like this:

```
class Array<T> {
  push(item: T): void;
  pop(): T;
  includes(item: T): boolean;
  length: number;
}
```

So if you have an array of strings (`Array<string>`), the type for i.e `push` "becomes" `push(item: string): void` :)

Reply:
I see! thanks everyone